public/app/plugins/datasource/prometheus/language_utils.ts

```diff
@@ -321,38 +322,49 @@ export interface PromLikeQuery extends DataQuery {
   range: boolean;
 }
 
+function getMaybeTokenStringContent(token: Token): string {
+  if (typeof token.content === 'string') {
+    return token.content;
+  }
+
+  return '';
+}
+
 export function extractLabelMatchers(tokens: Array<string | Token>): AbstractLabelMatcher[] {
   const labelMatchers: AbstractLabelMatcher[] = [];
 
-  for (let prop in tokens) {
-    if (tokens[prop] instanceof Token) {
-      let token: Token = tokens[prop] as Token;
-      if (token.type === 'context-labels') {
-        let labelKey = '';
-        let labelValue = '';
-        let labelOperator = '';
-        let contentTokens: any[] = token.content as any[];
-        for (let currentToken in contentTokens) {
-          if (typeof contentTokens[currentToken] === 'string') {
-            let currentStr: string;
-            currentStr = contentTokens[currentToken] as string;
-            if (currentStr === '=' || currentStr === '!=' || currentStr === '=~' || currentStr === '!~') {
-              labelOperator = currentStr;
-            }
-          } else if (contentTokens[currentToken] instanceof Token) {
-            switch (contentTokens[currentToken].type) {
-              case 'label-key':
-                labelKey = contentTokens[currentToken].content as string;
-                break;
-              case 'label-value':
-                labelValue = contentTokens[currentToken].content as string;
-                labelValue = labelValue.substring(1, labelValue.length - 1);
-                const labelComparator = FromPromLikeMap[labelOperator];
-                if (labelComparator) {
-                  labelMatchers.push({ name: labelKey, operator: labelComparator, value: labelValue });
-                }
-                break;
-            }
+  for (const token of tokens) {
+    if (!(token instanceof Token)) {
+      continue;
+    }
+
+    if (token.type === 'context-labels') {
+      let labelKey = '';
+      let labelValue = '';
+      let labelOperator = '';
+
+      const contentTokens = Array.isArray(token.content) ? token.content : [token.content];
+
+      for (let currentToken of contentTokens) {
+        if (typeof currentToken === 'string') {
+          let currentStr: string;
+          currentStr = currentToken;
+          if (currentStr === '=' || currentStr === '!=' || currentStr === '=~' || currentStr === '!~') {
+            labelOperator = currentStr;
+          }
+        } else if (currentToken instanceof Token) {
+          switch (currentToken.type) {
+            case 'label-key':
+              labelKey = getMaybeTokenStringContent(currentToken);
```

Comment:
This is probably the biggest runtime change. Previously this would potentially assign an array or object into a property expecting a string, which would (potentially, according to types) lead to runtime errors.

Now, it safely makes sure it's the correct type, otherwise sets it to an empty string.

Alternatively, we could either throw  an exception, or continue to iterate if it's not the expected type.